internal/migrations/migrator.go

```diff
@@ -153,9 +161,26 @@ func SetupMigrator(ctx context.Context, db Database, snapshotter Snapshotter, ac
 	}
 	migrator.lastChangeset = height
 
+	// Check if the migration is in progress or completed
+	if migrator.activeMigration != nil {
+		if migrator.lastChangeset >= migrator.activeMigration.StartHeight {
+			// migration is in progress
+			migrator.inProgress = true
+		}
+
+		if migrator.lastChangeset >= migrator.activeMigration.EndHeight-1 {
+			// migration is completed
+			migrator.halted = true
+		}
+	}
+
 	return migrator, nil
 }
 
+func (m *Migrator) SetCometClient(client *local.Local) {
+	m.cometClient = client
```

Comment:
Oh, but if you did go this way, I think we'd not bake a node client into it so explicitly, rather provide some kind of `ConsensusParamGetter` interface instance.  So it could be a getter from any more appropriate or direct source rather than going straight to cometbft.